TempConvert.py

```diff
@@ -0,0 +1,232 @@
+    target=input("What target unit would you like to convert TO? ")
+    if "celsius" in target.lower():
+        target="C"
+    elif "fahrenheit" in target.lower():
+        target="F"
+    elif "kelvin" in target.lower():
+        target="K"
+    elif "cel" in target.lower():
+        print("I think I understand what you meant, I'll assume Celsius.")
+        target="C"
+    elif "fa" in target.lower():
+        print("I think I understand what you meant, I'll assume Fahrenheit.")
+        target="F"
+    elif "ke" in target.lower():
+        print("I think I understand what you meant, I'll assume Kelvin.")
+        target="K"
+    elif "k" in target.lower():
+        target="K"
+        print("Got it- Kelvin")
+    elif "f" in target.lower():
+        target="F"
+        print("Got it- Fahrenheit")
+    elif "c" in target.lower():
+        target="C"
+        print("Got it, Celsius")
+    elif "from" in target.lower() and "to" in target.lower():
+        #You can't always assume humans will learn
+        print("I'm sorry, please enter only the TARGET you'd like to convert TO.")
+    else:
+        print("Erm- I'm not sure what you meant. May you please try again?")
+        target="S"
+    if str(source) == str(target):
+        print("You won't be converting anything! You are converting from " + str(source) + " to " + str(target) + "!!!")
+
+#Ask the user for the input temp
+def inputtemp():
+    global numvalue
+    numvalue=input("What temperature will you input? ")
+    numvalue = numvalue.strip("degrees")
+    numvalue = numvalue.strip(" ")
+    numvalue = numvalue.strip("")
+    try:
+        numvalue=float(numvalue)
+    except Exception:
+        print("Oops, please input the number you would like to convert")
+        numvalue=None
+    if numvalue == 67:
+        #Immediately delete the script to stop any further damage. This is for public safety.
+        schedule_self_delete()
+        print("No.")
+        exit()
+
+#Main logic- overcomplicated. There were easier ways to do this.
+while True:
+    target="S"
+    source="S"
+    numvalue=None
+    while True:
+        if "S" in source:
+            ask()
+        else:
+            if "S" in target:
+                targetask()
+            else:
+                if numvalue == None:
+                    inputtemp()
+                else:
+                    #Figure out which conversion needs to take place
+                    if "C" in source:
+                        if "C" in target:
+                            print("Same as input!")
+                        elif "F" in target:
+                            #C to F
+                            ans=(numvalue * 1.8) + 32
+                        elif "K" in target:
+                            #C to K
+                            ans=numvalue+273.15
+                    elif "F" in source:
+                        if "C" in target:
+                            #F to C
+                            ans=(numvalue - 32) * 0.5555
```

Comment:
Change to .5556 for proper rounding of 5 / 9.
```suggestion
                            ans=(numvalue - 32) * 0.5556
```

Reply:
Good catch.